pkg/engine/template_generator.go

```diff
@@ -692,6 +692,44 @@ func getContainerServiceFuncMap(cs *api.ContainerService) template.FuncMap {
 		"IsDockerContainerRuntime": func() bool {
 			return cs.Properties.OrchestratorProfile.KubernetesConfig.ContainerRuntime == api.Docker
 		},
+		"HasValidDockerConfig": func() bool {
+			_, err := common.GetDockerConfig(cs.Properties.OrchestratorProfile.KubernetesConfig.ContainerRuntimeConfig, nil)
+			return err == nil
+		},
+		"HasValidContainerdConfig": func() bool {
+			_, err := common.GetContainerdConfig(cs.Properties.OrchestratorProfile.KubernetesConfig.ContainerRuntimeConfig, nil)
+			return err == nil
+		},
+		"GetDockerConfig": func() string {
+			var overrides []func(*common.DockerConfig) error
+
+			if cs.Properties.HasNSeriesSKU() {
```

Comment:
I already squashed but L704-709 were not present which failed N series VMs with GPUs before this